zookeeper3/src/test/java/com/linecorp/armeria/common/zookeeper/ZooKeeperTestUtil.java

```diff
@@ -19,50 +19,57 @@
 import static org.assertj.core.api.Assertions.assertThatCode;
 import static org.awaitility.Awaitility.await;
 
-import java.io.IOException;
-import java.net.InetSocketAddress;
-import java.net.ServerSocket;
 import java.time.Duration;
 import java.util.List;
 import java.util.Random;
-import java.util.concurrent.ThreadLocalRandom;
 
 import com.google.common.collect.ImmutableList;
-import com.google.common.collect.ImmutableList.Builder;
 
 import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.internal.common.util.PortUtil;
 import com.linecorp.armeria.server.Server;
 
 public final class ZooKeeperTestUtil {
 
     private static final Random random = new Random();
 
     public static List<Endpoint> sampleEndpoints(int count) {
-        final int[] ports = unusedPorts(count);
-        final Builder<Endpoint> builder = ImmutableList.builder();
+        final int[] ports = unusedTcpPorts(count);
+        final ImmutableList.Builder<Endpoint> builder = ImmutableList.builder();
         for (int i = 0; i < count; i++) {
             builder.add(Endpoint.of("127.0.0.1", ports[i]).withWeight(random.nextInt(10000) + 1));
         }
         return builder.build();
     }
 
-    private static int[] unusedPorts(int numPorts) {
+    public static int[] unusedTcpPorts(int numPorts) {
```

Comment:
Could be package-private + `@VisibleForTesting`?

Reply:
As it is already in test, let me remove `public` only.